lib/ecto/log_entry.ex

```diff
@@ -89,4 +103,45 @@ defmodule Ecto.LogEntry do
       []
     end
   end
+
+  defp colorize({entry, data}) do
+    color = entry
+    |> sql_color
+    |> to_ansicolor
+
+    # We need to prepend the list with the color
+    # and append a clear color to reset.
+    data = data
+    |> Enum.reverse
+    |> Enum.concat([color])
+    |> Enum.reverse
+    |> Enum.concat([to_ansicolor(:clear)])
+
+    {entry, data}
+  end
+
+  defp sql_color(%{query: query}) do
+    match = &String.match?(query, &1)
+
+    cond do
+      match.(~r/\A\s*rollback/i) -> :red
+      match.(~r/\A\s*lock/i) -> :white
+      match.(~r/\A\s*select/i) -> :cyan
+      match.(~r/\A\s*insert/i) -> :green
+      match.(~r/\A\s*update/i) -> :yellow
+      match.(~r/\A\s*delete/i) -> :red
+      match.(~r/\A\s*begin/i) -> :magenta
+      match.(~r/\A\s*commit/i) -> :magenta
+      true -> :blue
+    end
+  end
+
+  defp to_ansicolor(:clear), do: "\e[0m"
```

Comment:
You should use `IO.ANSI.format [:red, original_string]` instead. This will:

1. Make sure it works properly on Windows
2. Abstract the color codes away from the code